docs/reference/data-streams/set-up-tsds.asciidoc

```diff
@@ -279,6 +279,21 @@ POST metrics-weather_sensors-dev/_rollover
 // TEST[setup:tsds]
 // TEST[teardown:tsds_cleanup]
 
+[discrete]
+[[set-up-component-templates]]
+==== A note about component templates and index.mode setting
+
+Configuring a TSDS via an index templates that uses component templates is a bit more complicated.
+With component templates typically mappings and settings get scattered across multiple component templates.
+When configuring the `index.mode` setting in a component template, then the index.routing_path setting needs to
+be defined as well in the same component template. Additionally the fields mentioned in the `index.routing_path`
+also need to be defined in the same component template with the `time_series_dimension` attribute enabled.
+
+The reasons for this is that each component template needs to be valid on its own and the time series index mode
+requires the `index.routing_path` setting. When configuring the  `index.mode` setting in an index template, then
+the `index.routing_path` setting is configured automatically. The `index.routing_path` setting is then derived  from
+the field mappings with `time_series_dimension` attribute enabled.
```

Comment:
So the reason it fails is that we don't see mappings when component template is created? Should we add a sentence to explain that here?

Reply:
No, the logic that automatically defines the routing_path setting is only activated with index templates. The main reason for that is then there is complete view of all the mappings. We could also do this component templates, but this doesn't happen today.